lib/Transforms/Instrumentation/ShadowMemSeaDsa.cc

```diff
@@ -1353,15 +1355,60 @@ void ShadowDsaImpl::solveUses(Function &F) {
                                        << " use(s) solved.\n");
 }
 
-} // namespace
+/// \brief Returns id of a field pointed to by the given cell \c
+unsigned ShadowDsaImpl::getFieldId(const dsa::Cell &c) {
+  assert(c.getNode());
+  unsigned id = m_maxId; // in case allocation of new id for the node is needed
+
+  const sea_dsa::Node *n = c.getNode();
+  unsigned offset = c.getOffset();
+  auto it = m_nodeIds.find(n);
+  if (it != m_nodeIds.end()) {
+    id = it->second + offset;
+  }
+  else{
+    m_nodeIds[n] = id;
+    if (n->size() == 0) {
+      // XXX: what does it mean for a node to have 0 size?
```

Comment:
all properties of dsa-nodes are by use. So if there is a pointer that appears in the program but is never used (i.e., read/stored) then the size is 0.

Size is also 0 if a node is collapsed -- we are not sure what size it is at all.

Reply:
Thanks! I added a comment for this!